chingu-voyages/v51-tier3-team-33
# Feature/connect auth to database

# Description
The goal here was to have logged in users through google be added to the database. Without this, there is no way to keep track of who did what within the app.

## Related to the following Jira/Trello issue 
N/A?

- Insert link to a relevant task/issue in Jira/Trello
N/A? 

## Main changes explained:
Updated User model to have an image field, so the images from google profiles will be saved to the database.
Updated the sign-in method in auth.config.ts to handle adding the users info to the database.
Added the session method in auth.config.ts to add the id of existing users in the database to the session object.

## How to test: 
1. Create feature/connect-authToDatabase branch
2. Pull changes from github.
3. Do `pnpm install` and `pnpm run dev` to run this PR locally
4. go to the "/login" url and login with google. 
5. check terminal to verify creation of new User object in the database.
6. go to /protected to sign out.
7. go back to /login to login 
8. check terminal to verify that the user already exists, should see User object.
9. check console logs in browser window to verify that user object includes the id of your user in the database.


## Note:
There are wonky warnings that I am not sure about. It will claim the find one method in the User model is not a function, same with the db connect method in dbConnect.ts, yet it will still work as intended. It makes a handful of calls to api/auth/session. I do not know if that is normal or a bug.
